### PR TITLE
63 boat port data mismatch

### PR DESCRIPTION
**Pull request by ckulfan**
Added a cache to use when assigning values to the port data structure because sometimes both boats on the same route can be going to the same direction. This ensures that we use the data from the boat at the dock for departure information and the lowest ETA from the boat heading to a new port.

Also changed the default value for AtDock to be false to ensure we do not return null values. Since we don't set this on one of the two ports for  routes that only have a single boat.

### CODE REVIEW

File: src/FerryTempo.js [hunk 1]
```diff
@@ -156,12 +158,31 @@ export default {
...
+         * (which means we do not have an ETA set).
+         */
+        let routeDirKey = routeAbbreviation + direction;
+        if (!routeDirCache[routeDirKey]) {
+          targetRoute['portData'][departingPort].BoatAtDock =  AtDock;
+          targetRoute['portData'][departingPort].PortDepartureDelay = boatDelay;
+          targetRoute['portData'][departingPort].PortStopTimer = timeAtDock;
+          targetRoute['portData'][arrivingPort].PortArrivalTimeMinus = arrivalTimeEta;
+          targetRoute['portData'][arrivingPort].PortETA = epochEta;
+          routeDirCache[routeDirKey] = true;
```
Comment (alexmarc-us) [inline src/FerryTempo.js:174]: Do we need logic for clearing the `routeDirCache`?

**Review by alexmarc-us — CHANGES_REQUESTED** [review]
Overall looks good, but I want to check that we are resetting the `routeDirCache` correctly. Does it need to be cleared at some point?

File: src/FerryTempo.js (hunk 1 above)
Comment (ckulfan) [reply src/FerryTempo.js:174]: The scope of the cache is within the for loop, so it is destroyed after we process all vessel location records.

**Review by alexmarc-us — APPROVED** [review]
LGTM

**Pull request merged**